### DISCUSSION TITLE
Update Mail SesTransport to use SesV2Client

### DISCUSSION THREAD

**Discussion by fideloper** [laravel/framework#37869] [Ideas] [answered]
Currently the [SesTransport](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Mail/Transport/SesTransport.php) uses [`$ses->sendRawEmail()`](https://docs.aws.amazon.com/aws-sdk-php/v3/api/api-email-2010-12-01.html#sendrawemail)

I couldn't find a PR/issue/discussion related to this and was wondering if anyone knew any downsides to submitting a PR to use the newer `SesV2Client`.

My selfish reason for this is the V2 client supports some additional features in SES, such as enabling a managed deliverability service/dashboard that AWS offers.

The v2 client doesn't have a `sendRawEmail()` method, but instead we'd use the [`sendEmail()`](https://docs.aws.amazon.com/aws-sdk-php/v3/api/api-sesv2-2019-09-27.html#sendemail) method which allows you to send a raw email.

Looks like the work involved would be:

1. Ensure a `SesV2Client` object is passed from the [MailManager](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Mail/MailManager.php)
2. Update the `SesTransport` (or make a new one?) to use the V2 client
3. Update related tests that reference this client
4. Update documentation which references the `sendRawEmail()` method

> The `SesV2Client` is roughly 2 years old based on GitHub history of the PHP SDK - Committed in Nov, 2019

**Comment by fideloper** [comment]
Here's a working example:

```php
$ses = new SesV2Client([
    'region' => 'us-east-1',
    'version' => 'latest',
]);

$mime = <<<MSSG
Message-ID: <28e2428e9a846edc734fa9f0eac00196@swift.generated>
Date: Wed, 30 Jun 2021 20:40:04 +0000
Subject: Test Message
From: Chris <chris@my-verified-sending-domain.com>
To: chris@example.com
MIME-Version: 1.0
Content-Type: text/html; charset=utf-8
Content-Transfer-Encoding: quoted-printable

<p>The Content</p>

MSSG;


$result = $ses->sendEmail([
    'FromEmailAddress' => 'chris@my-verified-sending-domain.com', //key($message->getSender() ?: $message->getFrom()),
    'Content' => [
        'Raw' => [
            'Data' => $mime,
        ],
    ],
]);
```

**Comment by ankurk91** [comment, marked as answer]
It has been landed in Laravel 9 recently
